sdk/include/ddk/ntstrsafe.h

```diff
@@ -2825,6 +2885,121 @@ RtlStringLengthWorkerW(
     return Status;
 }
 
+
+NTSTRSAFEAPI
+RtlUnicodeStringValidate(PCUNICODE_STRING SourceString)
+{
+    if (SourceString->Length % sizeof(WCHAR) != 0 ||
+        SourceString->MaximumLength % sizeof(WCHAR) != 0 ||
+        SourceString->Length > SourceString->MaximumLength ||
+        SourceString->MaximumLength > NTSTRSAFE_UNICODE_STRING_MAX_CCH * sizeof(WCHAR) ||
+        (SourceString->Buffer == NULL && (SourceString->Length != 0 || SourceString->MaximumLength != 0)))
+    {
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    return STATUS_SUCCESS;
+}
+
+NTSTRSAFEAPI
+RtlpUnicodeStringValidateEx(PCUNICODE_STRING SourceString, STRSAFE_DWORD dwFlags)
+{
+    if (dwFlags & STRSAFE_IGNORE_NULLS)
+        return STATUS_SUCCESS;
+
+    return RtlUnicodeStringValidate(SourceString);
```

Comment:
It might be better to actually implement the RtlUnicodeStringValidate() as a special case of RtlUnicodeStringValidateEx().
Also, could you check whether all these may not be particular cases of the "[RtlValidateUnicodeString()](https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/rtl/unicode.c;hb=579a784e0431e5400dacaf39ee6d97b0a380ebe9#l2544)" function?

Reply:
I will review this also.